src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -1008,6 +1024,10 @@ PeerImp::onMessage (std::shared_ptr <protocol::TMProposeSet> const& m)
 {
     protocol::TMProposeSet& set = *m;
 
+    if (overlay_.setup().expire &&
+            set.has_hops() && ! slot_->cluster())
+        set.set_hops(set.hops() + 1);
+
```

Comment:
Can we replace `set_hops` with `increment_hops` and `clear_hops`? I don't know protocol buffers, so that may be more trouble than it's worth.